### PR TITLE
Initialized vms array for cases when vms was returned as undefined.

### PR DESCRIPTION
**Pull request by h-kataria**
When a service didnt have vms linked to it, vms was coming back as undefined, causing JS errors and not displaying proper message on screen.

https://bugzilla.redhat.com/show_bug.cgi?id=1349601
https://bugzilla.redhat.com/show_bug.cgi?id=1349876

@dclarizio please review

before:
before

before

after:
after1

after

### CODE REVIEW

**Comment by dclarizio** [comment]
@AparnaKarve please review

File: client/app/states/services/details/details.state.js [hunk 1]
```diff
@@ -54,24 +54,13 @@
...
-    vm.service = service;
-
-    vm.activate = activate;
-    vm.removeService = removeService;
-    vm.editServiceModal = editServiceModal;
-    vm.retireServiceNow = retireServiceNow;
-    vm.retireServiceLater = retireServiceLater;
-    vm.ownershipServiceModal = ownershipServiceModal;
-    vm.reconfigureService = reconfigureService;
+    if (typeof vm.service.vms === 'undefined') {
```
Comment (AparnaKarve) [inline client/app/states/services/details/details.state.js]: @h-kataria For checking `undefined`, we usually use - `angular.isDefined()`
So something like this -
```
if (!angular.isDefined(vm.service.vms))
```

**Comment by AparnaKarve** [comment]
@h-kataria One of the issues in the BZ was to change the text to  "Power state: Unknown" for VMs with unknown power state. Could we add that change in this PR?

**Comment by h-kataria** [comment]
@AparnaKarve please re-review addressed your comments, uploaded additional screen-shots

**Comment by AparnaKarve** [comment]
@h-kataria The screenshot below shows truncated "Power State: unknown" string.

screen shot 2016-07-01 at 11 20 44 am

**Comment by h-kataria** [comment]
@AparnaKarve fixed truncation of Power State info, please re-test
ssui_fix

@epwinchell please review change in https://github.com/ManageIQ/manageiq-ui-self_service/pull/67/commits/20bbe7c2b60abfca0ec137c1304dae1784df43be to fix truncation of text, this is a short term fix until we upgrade Patternfly.

**Comment by AparnaKarve** [comment]
Screenshot with 20bbe7c2b60abfca0ec137c1304dae1784df43be applied -

screen shot 2016-07-01 at 11 27 24 am

Confirmed that it's working now.

File: client/app/states/services/details/details.html [hunk 2]
```diff
@@ -186,12 +186,12 @@ <h2 translate>Virtual Machines ({{ ::vm.service.vms.length }})</h2>
                 <strong tooltip="{{'The last heartbeat received from the instances'|translate}}" tooltip-placement="bottom" translate>Last Scan</strong>&nbsp;{{ item.last_scan_on | date:'medium' }}
               </span>
             </div>
-            <div class="col-lg-1 col-md-2 hidden-sm hidden-xs">
+            <div class="col-xlg-1 col-md-2 hidden-sm hidden-xs">
```
Comment (epwinchell) [inline client/app/states/services/details/details.html]: col-xlg isn't a valid bootstrap class

**Comment by h-kataria** [comment]
@epwinchell marking it as WIP until @himdel is done with patternfly upgrade.

**Comment by himdel** [comment]
Patternfly upgrade done in https://github.com/ManageIQ/manageiq-ui-self_service/pull/73 .. so you can probably start rebasing on top of that one :)

**Comment by AparnaKarve** [comment]
@h-kataria Power State column truncation issue that I had earlier is now resolved.
So the changes are looking good now 👍

**Comment by chessbyte** [comment]
@h-kataria Running `git cherry-pick -x -m 1 3f6fbcf` on darga causes merge conflict.

Please create PR against Darga branch.  /cc @dclarizio

**Comment by h-kataria** [comment]
@chessbyte created PR against Darga https://github.com/ManageIQ/manageiq-ui-self_service/pull/78

**Comment by chessbyte** [comment]
Backported to Darga via #78

**Pull request merged**